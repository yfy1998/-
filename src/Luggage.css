.Lug{
    margin-left: 41%;
    margin-top: 5px;
    padding: 0px;
    width: 230px;
    height: 140px;
    border: 2px solid;
    border-color: rgb(55, 109, 91);
    border-radius: 5px;
}

.Lug-label{
    text-align: right;
    width: 40px;
    height: 25px;
    margin-right: 8px;
    display: inline-block;
}

.Lug-input{
    margin-top: 5px;
    border-radius: 3px;
}