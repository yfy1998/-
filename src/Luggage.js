import React, { Component } from 'react';
import "./Luggage.css";

class Luggage extends Component{
    constructor(props){
        super(props);
        this.handleChange=this.handleChange.bind(this);
    }

    render(){
        return(
            <div className="Lug">
                <label>第{this.props.index+1}件行李</label>
                <br />
                <label className="Lug-label">重量</label>
                <input 
                    className="Lug-input"
                    type="text" 
                    name='weight' 
                    value={this.props.item.weight} 
                    onChange={this.handleChange}
                />   
                <br />
                <label className="Lug-label">长</label>
                <input 
                    className="Lug-input"
                    type="text" 
                    name='long' 
                    value={this.props.item.long} 
                    onChange={this.handleChange}
                />   
                <br />
                <label className="Lug-label">宽</label>
                <input 
                    className="Lug-input"
                    type="text" 
                    name='width' 
                    value={this.props.item.width} 
                    onChange={this.handleChange}
                />   
                <br />
                <label className="Lug-label">高</label>
                <input 
                    className="Lug-input"
                    type="text" 
                    name='height' 
                    value={this.props.item.height} 
                    onChange={this.handleChange}
                />   
                <br />
            </div>
        )
    }

    handleChange(e){
        const { LugChange }=this.props;
        const { index }=this.props;
        const name=e.target.name;
        const value=Number(e.target.value);
        //console.log(name);
        //console.log(value);
        if(isNaN(value)){
            //alert("行李信息请输入大于0的数值！");
            return;
        }
        //console.log(name+value);
        LugChange(index,name,value);
    }
}

export default Luggage; 