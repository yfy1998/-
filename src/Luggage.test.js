import React from 'react';
import { shallow,render } from 'enzyme';
import Luggage from './Luggage'
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const props = {
    // Jest 提供的mock 函数
    LugChange: jest.fn( (e) => {
    }),
    item: {}
}

it('when input change,onChang() should be called', () => {
    const wrapper=shallow(<Luggage {...props}/>);
    wrapper.find('input').at(0).simulate('change', {
        target: {
          value: '5'
        }});
     expect(props.LugChange).toBeCalled();
  });

  it('when input change a,onChang() should be called', () => {
    const wrapper=shallow(<Luggage {...props}/>);
    wrapper.find('input').at(0).simulate('change', {
        target: {
          value: 'a'
        }});
     expect(props.LugChange).toBeCalled();
  });

  it('snapshot2', () => {
    const wrapper = render(<Luggage {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  