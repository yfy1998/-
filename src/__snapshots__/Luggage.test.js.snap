// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot2 1`] = `
<div
  class="Lug"
>
  <label>
    第NaN件行李
  </label>
  <br />
  <label
    class="Lug-label"
  >
    重量
  </label>
  <input
    class="Lug-input"
    name="weight"
    type="text"
  />
  <br />
  <label
    class="Lug-label"
  >
    长
  </label>
  <input
    class="Lug-input"
    name="long"
    type="text"
  />
  <br />
  <label
    class="Lug-label"
  >
    宽
  </label>
  <input
    class="Lug-input"
    name="width"
    type="text"
  />
  <br />
  <label
    class="Lug-label"
  >
    高
  </label>
  <input
    class="Lug-input"
    name="height"
    type="text"
  />
  <br />
</div>
`;
